Framework/DataHandling/src/LoadBBY.cpp

```diff
@@ -514,16 +514,12 @@ LoadBBY::createInstrument(ANSTO::Tar::File &tarFile,
 
   // extract log and hdf file
   const std::vector<std::string> &files = tarFile.files();
-
-  int64_t fileSize = 0;
-  for (auto itr = files.begin(); itr != files.end(); ++itr)
-    if (itr->rfind(".hdf") == itr->length() - 4) {
-      tarFile.select(itr->c_str());
-      fileSize = tarFile.selected_size();
-      break;
-    }
-
-  if (fileSize != 0) {
+  const auto file_it =
+      std::find_if(files.begin(), files.end(), [](const std::string &file) {
+        return file.rfind(".hdf") == file.length() - 4;
+      });
+  if (file_it != files.end()) {
```

Comment:
I'm assuming that `fileSize` is only used to check that a filename is found. Checking for an empty file seems a bit extreme.